src/resource-metadata/lambda.js

```diff
@@ -0,0 +1,214 @@
+    const limit = pLimit(10);
+
+    const results = (await traverse(listOfFunctions.Functions, (lambdaFunctionVersionLatest) => {
+        return limit(functionThrottle(async () => {
+            const functionName = lambdaFunctionVersionLatest.FunctionName
+            const lambdaFunction = await lambdaClient.send(new GetFunctionCommand({ FunctionName: functionName }))
+            const versions = await lambdaClient.send(new ListVersionsByFunctionCommand({ FunctionName: functionName }))
+            const aliases = await lambdaClient.send(new ListAliasesCommand({ FunctionName: functionName }))
+
+            const versionsToCollect = versions.Versions.filter((version, index) => {
+                return (index <= latestVersionsPerFunction) // Is either $LATEST or one of the latestVersionsPerFunction latest released versions // This relies on the fact that AWS returns the functions in latest -> oldest order
+                    || (aliases.Aliases.some(alias => version.Version === alias.FunctionVersion)) // has an alias
+            })
+
+            return {
+                functionResource: makeLambdaFunctionResource(lambdaFunction),
+                aliasResources: aliases.Aliases.map(alias => makeAliasResource(functionName, alias)),
+                versionsToCollect,
+            }
+        }))
+    }))
+    const functionResources = results.map(x => x.functionResource)
+    const aliasResources = results.flatMap(x => x.aliasResources)
+    const versionsToCollect = results.flatMap(x => x.versionsToCollect)
+
+    console.info("Collecting function version details")
+
+    const functionVersionResources = await traverse(versionsToCollect, (lambdaFunctionVersion) => {
+        return limit(functionVersionThrottle(async () => {
+            const functionNameForRequests = lambdaFunctionVersion.Version === "$LATEST" 
+                ? lambdaFunctionVersion.FunctionName
+                : `${lambdaFunctionVersion.FunctionName}:${lambdaFunctionVersion.Version}`
+
+            const eventSourceMappings = await lambdaClient.send(new ListEventSourceMappingsCommand({ FunctionName: functionNameForRequests }))
+            let maybePolicy = null
+            try {
+                maybePolicy = await lambdaClient.send(new GetPolicyCommand({ FunctionName: functionNameForRequests }))
+            } catch (e) {
+                // GetPolicyCommand results in an error if the lambda has no policy defined.
+                // Ignore this error, and proceed with maybePolicy = null
+            }
+            return makeLambdaFunctionVersionResource(lambdaFunctionVersion, eventSourceMappings, maybePolicy)
+        }))
+    })
+
+    const resources = [...functionResources, ...functionVersionResources, ...aliasResources]
+
+    resources.forEach(f =>
+        console.info(`Resource: ${JSON.stringify(f)}`)
+    )
+
+    return resources
+}
```

Comment:
Divide and conquer.
The final look of this function can be 
```suggestion
async function collectLambdaResources() {

    console.info("Collecting list of functions")

    const listOfFunctions = await lambdaClient.send(new ListFunctionsCommand({}))

    console.info("Collecting function details")

    const {
        functionResources,
        aliasResources,
        versionsToCollect
    } = await getsomethingsomethingIdkTheDomain(listOfFunctions);

    console.info("Collecting function version details")
    const functionVersionResources = await getFunctionVersionResources(versionsToCollect);

    const resources = [...functionResources, ...functionVersionResources, ...aliasResources]

    resources.forEach(f =>
        console.info(`Resource: ${JSON.stringify(f)}`)
    )

    return resources
}
```
The content of the function can be even split into two separate files. The below parsers can go to these files too.